src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -186,7 +186,7 @@ public void Dispose()
             }
         }
 
-        internal object CaptureDisposable(object service)
+        internal virtual object CaptureDisposable(object service)
```

Comment:
I could be convinced of the value of the test, but arguing it's more valuable then real benchmarks that are of course monitored for regressions is laughable. I understand we live in the real world though, and what I'm asking for could be a long time coming.

At the very least we should verify that this change doesn't move perf in the wrong direction by making this virtual. You are eliding the very (short) method you just made virtual. If in practice, this method still gets called far more often than it gets elided, the entire change might not be worth it if that means virtualizing the method.

Do we know if expression tree compilation will ever inline this method? Could that be changed by visualizing it? Even if CaptureDisposable was never inlined in the first place, doesn't mean making this virtual is free.